hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/AbstractSyncHoodieClient.java

```diff
@@ -98,6 +99,8 @@ public abstract void createTable(String tableName, MessageType storageSchema,
 
   public abstract void updatePartitionsToTable(String tableName, List<String> changedPartitions);
 
+  public abstract void close();
```

Comment:
can we make this implement java AutoCloseable? so this can be used in try-with-resources clause. and you don't need to declare this here.

Reply:
good catch